packages/cdktf/lib/iterator.ts

```diff
@@ -0,0 +1,198 @@
+  ): ListIterator {
+    // TODO: this could return different iterators depending on the type of the list
+    // for example it could return a NumberListIterator whose iterator.key would be a number
+    return new ListIterator(list);
+  }
+
+  /**
+   * Creates a new iterator from a map
+   */
+  public static fromMap(
+    map:
+      | ComplexMap
+      | { [key: string]: any }
+      | { [key: string]: string }
+      | { [key: string]: number }
+      | { [key: string]: boolean }
+  ): MapIterator {
+    return new MapIterator(map);
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a string
+   */
+  getString(attribute: string): string {
+    return Token.asString(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a number
+   */
+  getNumber(attribute: string): number {
+    return Token.asNumber(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a (string) list
+   */
+  getList(attribute: string): string[] {
+    return Token.asList(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a number list
+   */
+  getNumberList(attribute: string): number[] {
+    return Token.asNumberList(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a map
+   */
+  getMap(attribute: string): { [key: string]: any } {
+    return Token.asAnyMap(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a map of strings
+   */
+  getStringMap(attribute: string): { [key: string]: string } {
+    return Token.asStringMap(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a map of numbers
+   */
+  getNumberMap(attribute: string): { [key: string]: number } {
+    return Token.asNumberMap(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @param attribute name of the property to retrieve
+   * @returns the given attribute of the current item iterated over as a map of booleans
+   */
+  getBooleanMap(attribute: string): { [key: string]: boolean } {
+    return Token.asBooleanMap(propertyAccess(this._getValue(), [attribute]));
+  }
+
+  /**
+   * @internal
+   */
+  protected _getValue(): any {
+    return Lazy.anyValue(
```

Comment:
Could you add a comment why lazy is needed, I think I remember why but I'm certain I'll be unsure in half a year